Swiften/Parser/PayloadParsers/MIXJoinParser.h

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) 2017 Tarun Gupta
+ * Licensed under the simplified BSD license.
+ * See Documentation/Licenses/BSD-simplified.txt for more information.
+ */
+
+#pragma once
+
+#include <memory>
+
+#include <Swiften/Base/API.h>
+#include <Swiften/Base/Override.h>
+#include <Swiften/Elements/MIXJoin.h>
+#include <Swiften/Parser/GenericPayloadParser.h>
+
+namespace Swift {
+    class PayloadParserFactoryCollection;
+    class PayloadParser;
+
+    class SWIFTEN_API MIXJoinParser : public GenericPayloadParser<MIXJoin> {
+        public:
+            MIXJoinParser(PayloadParserFactoryCollection* parsers);
```

Comment:
This only currently only needs PayloadParserFactoryCollection so it can be passed to the MIXSubscribeParser instance when created. If neither needs this collection this parameter can go away I think.